### PR TITLE
cleanup/Remove legacy throttler

### PR DESCRIPTION
**Pull request by petioptrv**
**Before submitting this PR, please make sure**:

- [X] Your code builds clean without any errors or warnings
- [X] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Removes the remaining legacy throttler and its associated test cases.

**Tests performed by the developer**:
None

**Tips for QA testing**:
None

### CODE REVIEW

**Review by CrimsonJacket — DISMISSED** [review]
Changes LGTM

File: hummingbot/core/data_type/order_book_message.py [hunk 1]
```diff
@@ -79,7 +79,9 @@ def has_trade_id(self) -> bool:
         return self.type == OrderBookMessageType.TRADE
 
     def __eq__(self, other: "OrderBookMessage") -> bool:
-        if self.has_update_id and other.has_update_id:
+        if self.type != other.type:
+            return False
+        elif self.has_update_id and other.has_update_id:
             return self.update_id == other.update_id
         elif self.has_trade_id and other.has_trade_id:
             return self.trade_id == other.trade_id
```
Comment (aarmoa) [inline hummingbot/core/data_type/order_book_message.py:87]: I think this should be implemented using logic operators only instead of if/elif statements

**Review by aarmoa — APPROVED** [review]
The changes look good to me. I just added a comment in a particular function I think we could implement without using if statements.

**Review by aarmoa — APPROVED** [review]
I approve the last refactoring too

**Pull request merged**